core/src/main/java/org/gbif/pipelines/core/parsers/location/LocationMatcher.java

```diff
@@ -0,0 +1,177 @@
+package org.gbif.pipelines.core.parsers.location;
+
+import org.gbif.api.vocabulary.Country;
+import org.gbif.common.parsers.geospatial.LatLng;
+import org.gbif.dwc.terms.DwcTerm;
+import org.gbif.dwc.terms.Term;
+import org.gbif.pipelines.core.parsers.InterpretationIssue;
+import org.gbif.pipelines.core.parsers.ParsedField;
+import org.gbif.pipelines.core.parsers.legacy.CountryMaps;
+import org.gbif.pipelines.core.ws.HttpResponse;
+import org.gbif.pipelines.core.ws.client.geocode.GeocodeServiceClient;
+import org.gbif.pipelines.io.avro.IssueType;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.gbif.pipelines.core.parsers.location.CoordinatesValidator.isAntarctica;
+
+/**
+ * Matches the location fields related to Country and Coordinates to find possible mismatches.
+ */
+public class LocationMatcher {
```

Comment:
The logic of this class is hard to follow, can we split in 2 flows? parse when a country is specified and when it is not?